src/main/java/macrobase/analysis/classify/BatchingPercentileClassifier.java

```diff
@@ -26,32 +34,37 @@ public BatchingPercentileClassifier(MacroBaseConf conf, Iterator<Datum> input) {
     public BatchingPercentileClassifier(MacroBaseConf conf, Iterator<Datum> input, double percentile) {
         super(conf, input);
         this.targetPercentile = percentile;
+        data = new ArrayList<>();
+
+        while(input.hasNext()) {
+            data.add(new DatumWithNorm(input.next()));
+        }
+        double[] scores = new double[data.size()];
+        for (int i = 0; i < data.size(); i++) {
+            scores[i] = data.get(i).getNorm();
+        }
+
+        Percentile pCalc = new Percentile().withNaNStrategy(NaNStrategy.MAXIMAL);
+        pCalc.setData(scores);
+        this.cutoff = pCalc.evaluate(scores, targetPercentile * 100);
+        log.info("{} Percentile Cutoff: {}", targetPercentile, cutoff);
```

Comment:
I thought knowing the rough distribution of percentiles was useful for almost any run, but maybe it's mostly useful for debugging runs. I'm fine with either, what's been your heuristic so far for deciding?